src_c/base.c

```diff
@@ -533,6 +533,103 @@ pg_TwoIntsFromObj(PyObject *obj, int *val1, int *val2)
     return 1;
 }
 
+static int
+pg_IntFromObjEx(PyObject *obj, int *val, const char *msg)
+{
+    if (PyFloat_Check(obj)) {
+        *val = (int)PyFloat_AS_DOUBLE(obj);
+        return 1;
+    }
+
+    *val = PyLong_AsLong(obj);
+    if (PyErr_Occurred()) {
+        if (msg && PyErr_ExceptionMatches(PyExc_TypeError)) {
+            PyErr_SetString(PyExc_TypeError, msg);
+        }
+        return 0;
+    }
+    return 1;
+}
+
+static int
+pg_IntFromSeqIndexEx(PyObject *obj, int index, int *val, const char *msg)
+
+{
+    int result = 0;
+    PyObject *item = PySequence_ITEM(obj, index);
```

Comment:
Won't this segfault if you pass it something that isn't a sequence? Since it's a macro.

Reply:
In theory yes, but as i've stated in the function docs(And i've been very clear), this function assumes the obj to be a sequence and it's even in the function name! no it's only natural for it to behave wrong if the developer uses this function the wrong way. They have been warned.

Reply:
Good work on the docs for these by the way, they look nice.

It seems like you've made this fast path function for the purposes of `pg_TwoIntsFromObjEx`, since it cuts out a redundant sequence check.

I'd rather not have a "public" C api function that can segfault on such an easy mistake to make, what do you think about an internal version of `pg_IntFromSeqIndexEx` just for `pg_TwoIntsFromObjEx`, as well as a public version that includes a sequence check?

Reply:
Thaks! yeah i think that could also work

Reply:
But again, many times one checks for `PySequence_Check()` anyway, so having a function like this check each time you wanna extract an element can be redundant.
For example it can happen you wanna extract 3 or 4 elements from a sequence. In that case if you add an extra check in `pg_IntFromSeqIndexEx` you'll end up adding 3-4 checks to your extractions, all redundant.